### ISSUE TITLE
unable to build any python examples 

### ISSUE THREAD

**Issue by LiyouZhou** [mvukov/rules_ros2#14]
commit : `106a7b615c6bcab0c96ea82d83c53351c6cb182e`
command: `bazel run //examples/chatter`
output: 
```
DEBUG: Rule 'ros2cli' indicated that a canonical reproducible form can be obtained by modifying arguments shallow_since = "1663423078 +0200"
DEBUG: Repository ros2cli instantiated at:
  /home/liyou/work/rules_ros2/WORKSPACE:5:18: in <toplevel>
  /home/liyou/work/rules_ros2/repositories/repositories.bzl:331:10: in ros2_repositories
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/utils.bzl:233:18: in maybe
Repository rule new_git_repository defined at:
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/git.bzl:186:37: in <toplevel>
DEBUG: Rule 'ros2_rosidl' indicated that a canonical reproducible form can be obtained by modifying arguments shallow_since = "1622892676 +0200"
DEBUG: Repository ros2_rosidl instantiated at:
  /home/liyou/work/rules_ros2/WORKSPACE:5:18: in <toplevel>
  /home/liyou/work/rules_ros2/repositories/repositories.bzl:45:10: in ros2_repositories
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/utils.bzl:233:18: in maybe
Repository rule new_git_repository defined at:
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/git.bzl:186:37: in <toplevel>
DEBUG: Rule 'ros2_rosidl_typesupport' indicated that a canonical reproducible form can be obtained by modifying arguments shallow_since = "1622921586 +0200"
DEBUG: Repository ros2_rosidl_typesupport instantiated at:
  /home/liyou/work/rules_ros2/WORKSPACE:5:18: in <toplevel>
  /home/liyou/work/rules_ros2/repositories/repositories.bzl:174:10: in ros2_repositories
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/utils.bzl:233:18: in maybe
Repository rule new_git_repository defined at:
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/git.bzl:186:37: in <toplevel>
DEBUG: Rule 'ros2_rcpputils' indicated that a canonical reproducible form can be obtained by modifying arguments shallow_since = "1622921172 +0200"
DEBUG: Repository ros2_rcpputils instantiated at:
  /home/liyou/work/rules_ros2/WORKSPACE:5:18: in <toplevel>
  /home/liyou/work/rules_ros2/repositories/repositories.bzl:70:10: in ros2_repositories
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/utils.bzl:233:18: in maybe
Repository rule new_git_repository defined at:
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/git.bzl:186:37: in <toplevel>
DEBUG: Rule 'ros2_rmw_implementation' indicated that a canonical reproducible form can be obtained by modifying arguments shallow_since = "1622921874 +0200"
DEBUG: Repository ros2_rmw_implementation instantiated at:
  /home/liyou/work/rules_ros2/WORKSPACE:5:18: in <toplevel>
  /home/liyou/work/rules_ros2/repositories/repositories.bzl:86:10: in ros2_repositories
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/utils.bzl:233:18: in maybe
Repository rule new_git_repository defined at:
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/git.bzl:186:37: in <toplevel>
DEBUG: Rule 'ros2_launch' indicated that a canonical reproducible form can be obtained by modifying arguments shallow_since = "1662324633 +0200"
DEBUG: Repository ros2_launch instantiated at:
  /home/liyou/work/rules_ros2/WORKSPACE:5:18: in <toplevel>
  /home/liyou/work/rules_ros2/repositories/repositories.bzl:306:10: in ros2_repositories
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/utils.bzl:233:18: in maybe
Repository rule new_git_repository defined at:
  /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/bazel_tools/tools/build_defs/repo/git.bzl:186:37: in <toplevel>
INFO: Analyzed target //examples/chatter:chatter (16 packages loaded, 1037 targets configured).
INFO: Found 1 target...
INFO: From Action external/ros2_rcl_interfaces/statistics_msgs/rosidl_adapter_map.idls:
Reading input file: external/ros2_rcl_interfaces/statistics_msgs/msg/MetricsMessage.msg
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/665/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/statistics_msgs/msg/MetricsMessage.idl
Reading input file: external/ros2_rcl_interfaces/statistics_msgs/msg/StatisticDataPoint.msg
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/665/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/statistics_msgs/msg/StatisticDataPoint.idl
Reading input file: external/ros2_rcl_interfaces/statistics_msgs/msg/StatisticDataType.msg
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/665/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/statistics_msgs/msg/StatisticDataType.idl
INFO: From Action external/ros2_rcl_interfaces/composition_interfaces/rosidl_adapter_map.idls:
Reading input file: external/ros2_rcl_interfaces/composition_interfaces/srv/ListNodes.srv
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/689/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/composition_interfaces/srv/ListNodes.idl
Reading input file: external/ros2_rcl_interfaces/composition_interfaces/srv/LoadNode.srv
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/689/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/composition_interfaces/srv/LoadNode.idl
Reading input file: external/ros2_rcl_interfaces/composition_interfaces/srv/UnloadNode.srv
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/689/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/composition_interfaces/srv/UnloadNode.idl
INFO: From Action external/ros2_rcl_interfaces/lifecycle_msgs/rosidl_adapter_map.idls:
Reading input file: external/ros2_rcl_interfaces/lifecycle_msgs/msg/State.msg
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/682/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/lifecycle_msgs/msg/State.idl
Reading input file: external/ros2_rcl_interfaces/lifecycle_msgs/msg/Transition.msg
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/682/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/lifecycle_msgs/msg/Transition.idl
Reading input file: external/ros2_rcl_interfaces/lifecycle_msgs/msg/TransitionDescription.msg
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/682/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/lifecycle_msgs/msg/TransitionDescription.idl
Reading input file: external/ros2_rcl_interfaces/lifecycle_msgs/msg/TransitionEvent.msg
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/682/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/lifecycle_msgs/msg/TransitionEvent.idl
Reading input file: external/ros2_rcl_interfaces/lifecycle_msgs/srv/ChangeState.srv
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/682/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/lifecycle_msgs/srv/ChangeState.idl
Reading input file: external/ros2_rcl_interfaces/lifecycle_msgs/srv/GetAvailableStates.srv
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/682/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/lifecycle_msgs/srv/GetAvailableStates.idl
Reading input file: external/ros2_rcl_interfaces/lifecycle_msgs/srv/GetAvailableTransitions.srv
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/682/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/lifecycle_msgs/srv/GetAvailableTransitions.idl
Reading input file: external/ros2_rcl_interfaces/lifecycle_msgs/srv/GetState.srv
Writing output file: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/682/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/lifecycle_msgs/srv/GetState.idl
ERROR: /home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/external/ros2_rcl_interfaces/BUILD.bazel:57:23: Action external/ros2_rcl_interfaces/rosgraph_msgs/msg/_clock.py failed: (Exit 1): rosidl_generator_py_app failed: error executing command bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app ... (remaining 2 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
ImportError in template 'external/ros2_rosidl_python/rosidl_generator_py/resource/_msg.py.em': 
Importing the multiarray numpy extension module failed.  Most
likely you are trying to import a failed build of numpy.
If you're working with a numpy git repo, try `git clean -xdf` (removes all
files not under version control).  Otherwise reinstall numpy.

Original error was: cannot import name 'multiarray' from partially initialized module 'numpy.core' (most likely due to a circular import) (/usr/lib/python3/dist-packages/numpy/core/__init__.py)

ImportError when expanding '_idl.py.em' into 'bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/rosgraph_msgs/msg/_clock.py': 
Importing the multiarray numpy extension module failed.  Most
likely you are trying to import a failed build of numpy.
If you're working with a numpy git repo, try `git clean -xdf` (removes all
files not under version control).  Otherwise reinstall numpy.

Original error was: cannot import name 'multiarray' from partially initialized module 'numpy.core' (most likely due to a circular import) (/usr/lib/python3/dist-packages/numpy/core/__init__.py)

Error processing idl file: bazel-out/k8-fastbuild/bin/external/ros2_rcl_interfaces/rosgraph_msgs/msg/Clock.idl
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/numpy/core/__init__.py", line 16, in <module>
    from . import multiarray
ImportError: cannot import name 'multiarray' from partially initialized module 'numpy.core' (most likely due to a circular import) (/usr/lib/python3/dist-packages/numpy/core/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/ros2_rosidl_python/rosidl_generator_py_app.py", line 40, in <module>
    sys.exit(main())
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/ros2_rosidl_python/rosidl_generator_py_app.py", line 36, in main
    return generate_py(args.generator_arguments_file, args.typesupport_impls.split(';'))
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/ros2_rosidl_python/rosidl_generator_py/rosidl_generator_py/generate_py_impl.py", line 61, in generate_py
    generate_files(generator_arguments_file, mapping)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/ros2_rosidl/rosidl_cmake/rosidl_cmake/__init__.py", line 96, in generate_files
    raise(e)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/ros2_rosidl/rosidl_cmake/rosidl_cmake/__init__.py", line 87, in generate_files
    expand_template(
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/ros2_rosidl/rosidl_cmake/rosidl_cmake/__init__.py", line 149, in expand_template
    interpreter.string(template_content, template_path, locals=data)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2391, in string
    self.safe(scanner, True, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2401, in safe
    self.parse(scanner, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2421, in parse
    token.run(self, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 1547, in run
    self.subrun(info[0][1], interpreter, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 1631, in subrun
    token.run(interpreter, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 1425, in run
    interpreter.execute(self.code, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2596, in execute
    _exec(statements, self.globals, locals)
  File "<string>", line 2, in <module>
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/ros2_rosidl/rosidl_cmake/rosidl_cmake/__init__.py", line 198, in _expand_template
    interpreter.string(content, str(template_path), kwargs)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2391, in string
    self.safe(scanner, True, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2401, in safe
    self.parse(scanner, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2421, in parse
    token.run(self, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 1425, in run
    interpreter.execute(self.code, locals)
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2596, in execute
    _exec(statements, self.globals, locals)
  File "<string>", line 3, in <module>
  File "/usr/lib/python3/dist-packages/numpy/__init__.py", line 142, in <module>
    from . import add_newdocs
  File "/usr/lib/python3/dist-packages/numpy/add_newdocs.py", line 13, in <module>
    from numpy.lib import add_newdoc
  File "/usr/lib/python3/dist-packages/numpy/lib/__init__.py", line 8, in <module>
    from .type_check import *
  File "/usr/lib/python3/dist-packages/numpy/lib/type_check.py", line 11, in <module>
    import numpy.core.numeric as _nx
  File "/usr/lib/python3/dist-packages/numpy/core/__init__.py", line 26, in <module>
    raise ImportError(msg)
ImportError: 
Importing the multiarray numpy extension module failed.  Most
likely you are trying to import a failed build of numpy.
If you're working with a numpy git repo, try `git clean -xdf` (removes all
files not under version control).  Otherwise reinstall numpy.

Original error was: cannot import name 'multiarray' from partially initialized module 'numpy.core' (most likely due to a circular import) (/usr/lib/python3/dist-packages/numpy/core/__init__.py)

Exception ignored in: <function Interpreter.__del__ at 0x7f038772c040>
Traceback (most recent call last):
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2094, in __del__
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2157, in shutdown
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2647, in finalize
  File "/home/liyou/.cache/bazel/_bazel_liyou/e75e8ec54e95eaef5eea312839932295/sandbox/linux-sandbox/674/execroot/com_github_mvukov_rules_ros2/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app.runfiles/rules_ros2_pip_deps_empy/site-packages/em.py", line 2200, in push
AttributeError: '_io.TextIOWrapper' object has no attribute 'push'
Target //examples/chatter:chatter failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /home/liyou/work/rules_ros2/examples/chatter/BUILD.bazel:38:12 Middleman _middlemen/examples_Schatter_Schatter-runfiles failed: (Exit 1): rosidl_generator_py_app failed: error executing command bazel-out/k8-opt-exec-2B5CBBC6/bin/external/ros2_rosidl_python/rosidl_generator_py_app ... (remaining 2 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
INFO: Elapsed time: 49.644s, Critical Path: 1.53s
INFO: 108 processes: 81 internal, 27 linux-sandbox.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```

**Comment by LiyouZhou** [comment]
It looks like python3.8 is trying to import an external version of numpy. however when I removed the system number package, the error became `ModuleNotFoundError: No module named 'numpy'`

**Comment by mvukov** [comment]
thanks for reporting this. It's time to set up a proper CI for the repo... The bug is fixed in https://github.com/mvukov/rules_ros2/commit/20b8fdf188bec0f8a761a967e47749abedecefd2. Please let me know if this solves issues for you.

**Comment by mvukov** [comment]
And another fix is in [921a483](https://github.com/mvukov/rules_ros2/commit/921a4836a39304b85c22ce7fcbb3b03a341e65de).
